hack/e2e/ensure_htpasswd.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -eux
+
+# Check if htpasswd is installed and install it if not
+verify_htpasswd_version()
+{
+    if ! [ -x "$(command -v htpasswd)" ]; then
+        echo "htpasswd could not be found, installing..."
+        sudo apt-get update
+        sudo apt-get install -y apache2-utils
```

Comment:
not specific to this PR, but generally these `hack/e2e/` scripts wont work locally for a mac user like me. I will add an issue to make it work for other architectures as well. 
cc @lentzi90 @maxrantil

Reply:
Yes it would be great to make this more flexible so it can work also for development and not just CI. So far my focus has been to just set up the CI, but for the long term I definitely intend to make this easier for developers to use locally.
I would ask that you do not make the CI unnecessarily complicated just to accommodate mac users though :wink:

Reply:
I dis-asgree here. My understanding was the `hack/ `scripts are also used to test the CI test locally. If `hack/e2e` scripts are not meant to be used by developers locally, then I can agree. 

> I would ask that you do not make the CI unnecessarily complicated just to accommodate mac users though 😉

Reply:
Anyways, I have created an issue here anyhow https://github.com/metal3-io/baremetal-operator/issues/1487, lets discuss it on the issue.
/approve